### PR TITLE
chore: Add newrelic tracking spans to AppViewer engine and AppEditor engine

### PR DESCRIPTION
**Pull request by dvj1988**
## Description
Add new relic tracking spans to all fn calls in Appviewer and AppEditor engine.

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Sanity"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9868467158>
> Commit: 9267930a1501b1a6eaab985c714c04a4f2a5da52
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9868467158&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Sanity`
> Spec:
> <hr>Wed, 10 Jul 2024 05:32:52 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Enhanced app performance tracking with added telemetry spans for key functions.

- **Tests**
  - Updated test cases to include telemetry span tracking, ensuring better performance monitoring and validation.

### CODE REVIEW

File: app/client/src/entities/Engine/index.ts [hunk 1]
```diff
@@ -97,25 +112,44 @@ export default abstract class AppEngine {
...
+
       if (!this._urlRedirect) return;
       const newURL: string = yield call(
         this._urlRedirect.generateRedirectURL.bind(this),
         pageId,
         pageIdInUrl,
       );
+
+      endSpan(loadAppUrlSpan);
+
```
Comment (vsvamsi1) [inline app/client/src/entities/Engine/index.ts:152]: Is this necessary, no blocking or expensive code here?

Comment (dvj1988) [reply app/client/src/entities/Engine/index.ts:152]: yeah that's true, I just added them to all components.

Comment (dvj1988) [reply app/client/src/entities/Engine/index.ts:152]: loadAppData set the value of `_urlRedirect` and it is called before `loadAppURL` in initSagas. This block is present for just type safety.

But yeah I agree, we should avoid any case where there are spans that don't end.

**Review by vsvamsi1 — APPROVED** [review]

**Review by vsvamsi1 — DISMISSED** [review]

**Review by vsvamsi1 — APPROVED** [review]

**Pull request merged**